Kernel/Arch/riscv64/Interrupts.cpp

```diff
@@ -32,18 +37,80 @@ ErrorOr<u8> reserve_interrupt_handlers(u8)
     TODO_RISCV64();
 }
 
-void register_generic_interrupt_handler(u8, GenericInterruptHandler&)
+static void revert_to_unused_handler(u8 interrupt_number)
 {
-    TODO_RISCV64();
+    auto* handler = new UnhandledInterruptHandler(interrupt_number);
+    handler->register_interrupt_handler();
+}
+
+// FIXME: Share the code below with Arch/{x86_64,aarch64}/Interrupts.cpp
+void register_generic_interrupt_handler(u8 interrupt_number, GenericInterruptHandler& handler)
+{
+    auto*& handler_slot = s_interrupt_handlers[interrupt_number];
+    if (handler_slot == nullptr) {
+        handler_slot = &handler;
+        return;
+    }
+    if (handler_slot->type() == HandlerType::UnhandledInterruptHandler) {
+        auto* unhandled_handler = static_cast<UnhandledInterruptHandler*>(handler_slot);
+        unhandled_handler->unregister_interrupt_handler();
+        delete unhandled_handler;
+        handler_slot = &handler;
+        return;
```

Comment:
yeesh, I see why there's a fixme above at the declaration of s_interrupt_handlers. no change requested since this is probably copy pasta..